### DISCUSSION TITLE
!!! Urgent please help! Choppy Motion - Gaps in Linear Motion Path?

### DISCUSSION THREAD

**Discussion by invishane** [keiwando/toonsquid#524] [General]
hey guys I am moving a background in a straight line and I’m getting these gaps in the linear motion path, it’s making the animation really choppy when it hits these paths - any ideas on how to fix?? I’m probably missing something so simple, thanks for any help! 

![IMG_6137](https://github.com/keiwando/toonsquid/assets/64921279/588c5e7d-b7ae-4232-9026-b16e023f0ff9)
![IMG_6138](https://github.com/keiwando/toonsquid/assets/64921279/e103eb60-4f6e-4090-898f-07fc93859204)

**Comment by DinoAwakening** [comment]
> hey guys I am moving a background in a straight line and I’m getting these gaps in the linear motion path, it’s making the animation really choppy when it hits these paths - any ideas on how to fix?? I’m probably missing something so simple, thanks for any help!
> 
> ![IMG_6137](https://private-user-images.githubusercontent.com/64921279/301231628-588c5e7d-b7ae-4232-9026-b16e023f0ff9.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NDExMTcsIm5iZiI6MTcwNjc0MDgxNywicGF0aCI6Ii82NDkyMTI3OS8zMDEyMzE2MjgtNTg4YzVlN2QtYjdhZS00MjMyLTkwMjYtYjE2ZTAyM2YwZmY5LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMVQyMjQwMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYWY4NDNhNzEzMmVlODc3ZGZlMmIyMzY3ODIzZmUzZWY1MDJiNTIzOGUyNGNkMDk5ODg3YTdiNGJkZjM1MWIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UEDt7CruwYf65u-ZKVP1EqimcjJA6XF-Tu57Rb4o3PA) ![IMG_6138](https://private-user-images.githubusercontent.com/64921279/301231643-e103eb60-4f6e-4090-898f-07fc93859204.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NDExMTcsIm5iZiI6MTcwNjc0MDgxNywicGF0aCI6Ii82NDkyMTI3OS8zMDEyMzE2NDMtZTEwM2ViNjAtNGY2ZS00MDkwLTg5OGYtMDdmYzkzODU5MjA0LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMVQyMjQwMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OGE3Y2EwNGI5OTcyNzM0MDQ4MzFhMWIzOTQ3MTkxNWRhMWFjNjViZTRkMjU1ZjFkMjQ4ZjI4MTZlOGRiNTA5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gQon2JBCm1Nivd2UStGuNIACC_iadfklupHMSnNeSwU)

Either you ran out of RAM, or your file is corrupted. Try making a new project, do that exact same motion, for the exact same period of time you need it. Then, copy the whole animation layer, and paste it in the project you actually need it. Hope it works!

(I still believe it’s your RAM suffering tho…)

**Comment by Cat-Butler** [comment]
@DinoAwakening might have it correct: also worth doing a hard restart to clear out any memory glitches, and also checking you still have at least 15% or so of your storage free to work with (the iPad Settings app > General > iPad Storage).

I know you said "linear motion path", but would you mind checking easing?

1. Tap to select/highlight the first Position keyframe for that moving background layer on the timeline.
2. With that keyframe selected (the first little Position diamond on the timeline should turn blue/cyan), the interpolation control should appear underneath the timeline (looks like a thin stretched S shape). Tap it to bring up the Easing Curve window.
3. Check that it is correctly set to Linear (If it accidentally got set to Bounce, you might see results similar to what you described).

**Comment by invishane** [comment]
Hey!  first I want to thank you guys who responded this community is awesome and your responses were very informative. So I checked, I had about 1.65TB free on my M2 iPad, I cleared cache and storage in the project and swiped away every open app and restated the iPad, made a new project with ONLY the moving background, I play with the easing settings all the time so I made sure they were linear, and I got the same result, it seems like if the two points are too far apart it can’t do a consistent smooth tween. BUT I tried placing key frames down as I scrubbed across the timeline and it worked to smooth it out! The video is only about 1:30 seconds so I put the key frames about every 180 frames it’s WAY smoother now! 

Thanks again!

**Reply by Cat-Butler** [reply]
Thanks for the further detail and clarification, and for posting a workaround in case this affects any other users!

**Comment by keiwando** [comment]
I can reproduce this and will work on fixing this with a future update. Thanks for posting about it!

**Comment by keiwando** [comment]
This should now be fixed with the 1.10.8 update.

**Reply by steryo** [reply]
Experiencing the same issue on every easing method; iPad pro m1 11inch

**Reply by keiwando** [reply]
Could you please send me a copy of a project file with this issue to contact@toonsquid.com for me to take a closer look? If it's a specific layer, you can also copy and paste it into a new empty project so you don't have to send everything else from the original project.
